tools/Config.mk

```diff
@@ -577,9 +583,9 @@ $(1)_$(2):
 
 endef
 
-export DEFINE_PREFIX := $(subst X,,${shell $(DEFINE) "$(CC)" "X"})
-export INCDIR_PREFIX := $(subst "X",,${shell $(INCDIR) "$(CC)" "X"})
-export INCSYSDIR_PREFIX := $(subst "X",,${shell $(INCDIR) -s "$(CC)" "X"})
+export DEFINE_PREFIX ?= $(subst X,,${shell $(DEFINE) "$(CC)" "X" 2> ${EMPTYFILE}})
```

Comment:
does ?= still improve the build speed?

Reply:
Yes, ?= only assigns if the variable does not exist
https://stackoverflow.com/questions/4879592/whats-the-difference-between-and-in-makefile

The following is the data of different branch compilation times:
|   branch| first  |  second |  
|---|---|---|
| release-11  |  0m9.706s |  0m1.697s |  
|  master | 0m4.810s  |  0m0.998s |   
|  HEAD |   0m4.499s |  0m1.078s |

Reply:
OK